app/configurator/components/filters.tsx

```diff
@@ -354,44 +355,86 @@ const MultiFilterContent = ({
 
   return (
     <Box sx={{ position: "relative" }}>
       <Box mb={4}>
-        <Box sx={{ justifyContent: "space-between", display: "flex" }}>
+        <Box
+          sx={{
+            justifyContent: "space-between",
+            display: "flex",
+            flexDirection: "column",
+            alignItems: "flex-end",
+          }}
+        >
           {chartConfig.activeField === "segment" ? (
-            <FormControlLabel
-              componentsProps={{ typography: { variant: "body2" } }}
-              control={<Switch {...interactiveFilterProps} />}
-              label={
-                <MaybeTooltip
-                  tooltipProps={{ enterDelay: 600 }}
-                  title={
-                    <TooltipTitle
-                      text={
-                        <Trans id="controls.filters.interactive.tooltip">
-                          Allow users to change filters
+            <>
+              <Flex
+                sx={{ height: "33px", alignItems: "center", width: "100%" }}
+              >
+                <FormControlLabel
+                  componentsProps={{ typography: { variant: "body2" } }}
+                  control={<Switch {...interactiveFilterProps} />}
+                  label={
+                    <MaybeTooltip
+                      tooltipProps={{ enterDelay: 600 }}
+                      title={
+                        <TooltipTitle
+                          text={
+                            <Trans id="controls.filters.interactive.tooltip">
+                              Allow users to change filters
+                            </Trans>
+                          }
+                        />
+                      }
+                    >
+                      <div>
+                        <Trans id="controls.filters.interactive.toggle">
+                          Interactive
                         </Trans>
+                      </div>
+                    </MaybeTooltip>
+                  }
+                />
+              </Flex>
+              <Flex
+                sx={{ height: "33px", alignItems: "center", width: "100%" }}
+              >
+                <FormControlLabel
+                  componentsProps={{ typography: { variant: "body2" } }}
+                  control={<Switch {...visibleLegendProps} />}
+                  label={
+                    <MaybeTooltip
+                      tooltipProps={{ enterDelay: 600 }}
+                      title={
+                        <TooltipTitle
+                          text={
+                            <Trans id="controls.filters.show-legend.tooltip">
+                              Allow users to change Legend visibility
+                            </Trans>
+                          }
+                        />
                       }
-                    />
+                    >
+                      <div>
+                        <Trans id="controls.filters.show-legend.toggle">
```

Comment:
Nit: Translation string should probably be named along the lines of `show-legend-title` 🤔